### PR TITLE
docs(snippets): punycode

### PR DESCRIPTION
**Pull request by EdamAme-x**
as title

### CODE REVIEW

**Comment by yusukebe** [comment]
@EdamAme-x 

Have you checked by launching the website app locally?

You should modify `.vitepress/config.ts`.

**Comment by EdamAme-x** [comment]
sry @yusukebe

File: .vitepress/config.ts [hunk 1]
```diff
@@ -138,6 +138,10 @@ const sidebarsSnippets = (): DefaultTheme.SidebarItem[] => [
         text: 'Grouping routes for RPC',
         link: '/snippets/grouping-routes-rpc',
       },
+      {
+        text: 'Use Unicode for url',
```
Comment (yusukebe) [inline .vitepress/config.ts:142]: `url` should be `URL`.

File: snippets/punycode.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+### Punycode
+Use Unicode for url
+
+## Snippets
+
+this example is for who want to use **Unicode** for url.  
+
+Currently, `c.redirect` and anyone do not support **Unicode**.
```
Comment (yusukebe) [inline snippets/punycode.md:8]: Do not break lines within a paragraph.

File: snippets/punycode.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+You should conversion **Punycode** from Unicode.  
+There are many such libraries, but you do not need to use them.  
+The **Web standard api** is sufficient.  
+
+```typescript
+const url = new URL("https://🔥.com").href
+console.log(url) // https://xn--4v8h.com
+```
+
+`c.redirect(new URL("https://🔥.com").href)`
```
Comment (yusukebe) [inline snippets/punycode.md:21]: Is this necessary?

File: snippets/punycode.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Snippets
+
+this example is for who want to use **Unicode** for url.  
+
+Currently, `c.redirect` and anyone do not support **Unicode**.  
+`c.redirect("https://🔥.com")` throw Error.  
+`https://🔥.com` is expressed like `https://xn--4v8h.com` in **Punycode**.  
+
+You should conversion **Punycode** from Unicode.
```
Comment (yusukebe) [inline snippets/punycode.md:12]: `conversion` should be `convert`.

File: snippets/punycode.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+this example is for who want to use **Unicode** for url.  
+
+Currently, `c.redirect` and anyone do not support **Unicode**.  
+`c.redirect("https://🔥.com")` throw Error.  
+`https://🔥.com` is expressed like `https://xn--4v8h.com` in **Punycode**.  
+
+You should conversion **Punycode** from Unicode.  
+There are many such libraries, but you do not need to use them.  
+The **Web standard api** is sufficient.
```
Comment (yusukebe) [inline snippets/punycode.md:14]: `Web Standard API` is better.

File: snippets/punycode.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+### Punycode
```
Comment (yusukebe) [inline snippets/punycode.md:1]: Should be `# Punycode`.

**Comment by yusukebe** [comment]
Hi @EdamAme-x !

I've left the comments. Check them please!

**Pull request closed without merging**